server/app/models/QuestionModel.java

```diff
@@ -106,15 +109,49 @@ public QuestionModel addVersion(VersionModel version) {
   }
 
   /** Populates column values from {@link QuestionDefinition}. */
-  @PreUpdate
   @PrePersist
+  public void persistQuestionDefinition() {
+    // Play Ebeans starting at v6.2.0 includes updated Ebeans that fixes a bug we
+    // had relied on to mark the json fields as dirty. We now need to manually
+    // trigger the dirty flag or the @PrePersist/@PreUpdate annotations don't
+    // get triggered.
+    io.ebean.DB.markAsDirty(this);
+    setFieldsFromQuestionDefinition(questionDefinition);
+    // If this is a new model and the concurrency token isn't set, create one.
+    if (this.concurrencyToken == null) {
+      this.concurrencyToken = UUID.randomUUID();
+    }
+  }
+
+  /** Populates column values from {@link QuestionDefinition}. */
+  @PreUpdate
   public void persistChangesToQuestionDefinition() {
     // Play Ebeans starting at v6.2.0 includes updated Ebeans that fixes a bug we
     // had relied on to mark the json fields as dirty. We now need to manually
     // trigger the dirty flag or the @PrePersist/@PreUpdate annotations don't
     // get triggered.
     io.ebean.DB.markAsDirty(this);
     setFieldsFromQuestionDefinition(questionDefinition);
+
+    // Verify the concurrency token is still valid before persisting
+    int numMatchingEntities =
+        this.db()
+            .find(QuestionModel.class)
+            .setLabel("QuestionModel.preUpdate")
+            .where()
+            .eq("id", this.id)
+            .eq("concurrency_token", this.concurrencyToken)
+            .findCount();
+
+    if (numMatchingEntities != 1) {
+      throw new ConcurrentUpdateException(
+          String.format(
+              "An entity with the concurrency_token %s could not be found.",
```

Comment:
(non-blocking) It would probably be good to clarify which entity's `concurrency_token` couldn't be found and the ID in order to aid tracking down the issue. I'm fine with that being a separate PR.

Reply:
done!